packages/cli/src/push/index.ts

```diff
@@ -85,14 +85,27 @@ export class Push extends TypedEmitter<PushEvents> {
 		this.backend.sendToAll(type, data);
 	}
 
+	/** Returns whether a given push ref is registered. */
+	hasPushRef(pushRef: string) {
+		return this.backend.hasPushRef(pushRef);
+	}
+
 	send<Type extends PushType>(type: Type, data: PushPayload<Type>, pushRef: string) {
-		/**
-		 * Multi-main setup: In a manual webhook execution, the main process that
-		 * handles a webhook might not be the same as the main process that created
-		 * the webhook. If so, the handler process commands the creator process to
-		 * relay the former's execution lifecycle events to the creator's frontend.
-		 */
-		if (this.instanceSettings.isMultiMain && !this.backend.hasPushRef(pushRef)) {
+		const { isWorker, isMultiMain } = this.instanceSettings;
+
+		if (isWorker || (isMultiMain && !this.hasPushRef(pushRef))) {
```

Comment:
I don't quite understand why main needs to re-relay these events? When worker publishes `relay-execution-lifecycle-event` event, it should go to all mains right? Why does a the other mains need to re-send them?

Reply:
Worker relaying and main relaying are for two entirely separate purposes, see comment above:

https://github.com/n8n-io/n8n/blob/c6d97ec24196182bef9af9126a9cd8761b2b9113/packages/cli/src/push/index.ts#L96-L107